### PR TITLE
changed to use resource bundles

### PR DESCRIPTION
**Pull request by edewit**

### CODE REVIEW

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_ca.properties [hunk 1]
```diff
@@ -0,0 +1,275 @@
+#converted from translation.json
+#Thu Oct 26 10:13:57 CEST 2023
```
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_ca.properties:2]: I think we can remove these headers no? I don't think they'll provide much value for future reference.

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_it.properties
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_it.properties:1]: This file is empty, perhaps it can be removed? Seems weird though, we didn't have Italian translations before?

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_lv.properties
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_lv.properties:1]: Also empty.

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_nl.properties
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_nl.properties:1]: Also empty.

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_pl.properties
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_pl.properties:1]: Also empty.

File: js/apps/admin-ui/src/i18n/i18n.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import { createInstance } from "i18next";
 import { initReactI18next } from "react-i18next";
 
+import HttpBackend from "i18next-http-backend";
 import environment from "../environment";
 import { joinPath } from "../utils/joinPath";
-import { OverridesBackend } from "./OverridesBackend";
+
+type KeyValue = { key: string; value: string };
```
Comment (jonkoops) [inline js/apps/admin-ui/src/i18n/i18n.ts:8]: Perhaps we should put this type somewhere in a common section, we often repeat this exact type in other places. But not part of this PR.

File: js/apps/admin-ui/src/i18n/i18n.ts [hunk 3]
```diff
@@ -19,9 +21,19 @@ export const i18n = createInstance({
     escapeValue: false,
   },
   backend: {
-    loadPath: joinPath(environment.resourceUrl, `locales/{{lng}}/{{ns}}.json`),
+    loadPath: joinPath(
```
Comment (jonkoops) [inline js/apps/admin-ui/src/i18n/i18n.ts:24]: Couple of notes here:

1. This can be made ype-safe by using [the types]([https://github.com/i18next/i18next-http-backend#typescript) of the HTTP back-end.
2. The `KeyValue` type can be directly assigned to the result of `JSON.parse`.
3. Perhaps we can use `Object.fromEntries()` to build the result? Either way this approach is also fine by me.

For example:

```ts
createInstance({
  backend: {
    parse(data) {
      const messages: KeyValue[] = JSON.parse(data);

      return Object.fromEntries(
        messages.map(({ key, value }) => [key, value] as const),
      );
    },
  },
} satisfies InitOptions<HttpBackendOptions>);
```

File: js/apps/admin-ui/src/i18n/i18n.ts
Comment (jonkoops) [inline js/apps/admin-ui/src/i18n/i18n.ts:1]: I assume we'll want to do the same for the Account Console. Do we have an issue to track that?

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_lv.properties
Comment (edewit) [reply js/apps/admin-ui/maven-resources/theme-resources/messages/messages_lv.properties:1]: yes they always have been empty

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_pl.properties
Comment (edewit) [reply js/apps/admin-ui/maven-resources/theme-resources/messages/messages_pl.properties:1]: we could remove in the passed there have been PRs that added them empty

File: js/apps/admin-ui/src/i18n/i18n.ts (hunk 3 above)
Comment (edewit) [reply js/apps/admin-ui/src/i18n/i18n.ts:24]: That is pretty useless don't you think

File: js/apps/admin-ui/src/i18n/i18n.ts (hunk 2 above)
Comment (edewit) [reply js/apps/admin-ui/src/i18n/i18n.ts:8]: If it was the same key, value that would make sense but this are translation key value and not attribute key value

File: js/apps/admin-ui/src/i18n/i18n.ts
Comment (edewit) [reply js/apps/admin-ui/src/i18n/i18n.ts:1]: We have lots of issuer on the account console about how translations don't work

**Review by agagancarczyk — DISMISSED** [review]

File: js/apps/admin-ui/src/i18n/i18n.ts
Comment (jonkoops) [reply js/apps/admin-ui/src/i18n/i18n.ts:1]: Ok, then let's consider this as resolved.

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_ja.properties
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_ja.properties:1]: The Unicode characters in this file are escaped, rather than having their literal value, making this file impossible to read for humans. Likely this is because whatever was used in converting these files amuses the property files are encoded in ANSI rather than UTF-8.


These files in Keycloak however, are always interpreted as UTF-8, so we should be using the literal values instead. This should be applied to the other property files as well.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/src/i18n/i18n.ts (hunk 3 above)
Comment (jonkoops) [reply js/apps/admin-ui/src/i18n/i18n.ts:24]: No, I don't. It's just a small enhancement, do what you see fit.

File: js/apps/admin-ui/src/i18n/i18n.ts (hunk 2 above)
Comment (jonkoops) [reply js/apps/admin-ui/src/i18n/i18n.ts:8]: Yeah, I guess it doesn't hurt having an extra type. Consider this resolved.

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_es.properties
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_es.properties:1]: Still seeing some encoded Unicode characters in this file.

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_lt.properties
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_lt.properties:1]: Same for this one, still seeing some encoded Unicode characters.

File: js/apps/admin-ui/maven-resources/theme-resources/messages/messages_no.properties
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme-resources/messages/messages_no.properties:1]: Here as well, still seeing some encoded Unicode characters.

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Review by jonkoops — APPROVED** [review]

**Review by agagancarczyk — APPROVED** [review]
LGTM!

**Review by ssilvert — APPROVED** [review]
LGTM
